liquibase-core/src/main/java/liquibase/command/core/TagCommandStep.java

```diff
@@ -19,34 +31,53 @@ public class TagCommandStep extends AbstractCliWrapperCommandStep {
 
     static {
         CommandBuilder builder = new CommandBuilder(COMMAND_NAME);
-        URL_ARG = builder.argument(CommonArgumentNames.URL, String.class).required()
-                .description("The JDBC database connection URL").build();
+        TAG_ARG = builder.argument("tag", String.class).required()
+                .description("Tag to add to the database changelog table").build();
+        URL_ARG = builder.argument(CommonArgumentNames.URL, String.class).description("The JDBC database connection URL").build();
```

Comment:
URL should be a required argument.

Reply:
As it is used within Liquibase.tag now, I removed this requirement as it may pass the database as an argument (and it broke a lot of tests because of that).  But I added a runtime validation to AbstractCommandStep to validate URL requirement before using it and throw the same kind of exception.
What we really would need would be an "OR" validation: URL or database.